### PR TITLE
The common scripts of PR git diff changes.

### PR DESCRIPTION
**Pull request by sima-zhu**
Have a common script for git diff changes.
Test proposal:
PR triggered(Compliance credscan): https://github.com/Azure/azure-sdk-for-js/pull/20643
Manually triggered(Build print diff): https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1407323&view=results
Test on Add, Update, Delete, Rename cases(Compliance credscan): https://github.com/Azure/azure-sdk-for-js/pull/20643
Test on sparse-checkout cases(Build print diff): https://github.com/Azure/azure-sdk-for-java/pull/27335

image
The git diff changes will add the file outside of the sparse checkout

### CODE REVIEW

File: eng/common/scripts/git-diff-changes.ps1
Comment (benbp) [inline eng/common/scripts/git-diff-changes.ps1:20]: I imagine our link checker won't like the `/en/` segment in there

**Review by chidozieononiwu — APPROVED** [review]
Works for the eng/common workflow enforcer case

File: eng/common/scripts/git-diff-changes.ps1
Comment (benbp) [inline eng/common/scripts/git-diff-changes.ps1:35]:
```suggestion
if(!$changedFiles) {
    Write-Host "No changed files in git diff between $TargetCommittish and $SourceCommittish"
}
foreach ($file in $changedFiles) {
    Write-Host "The diff file is: $file"
}
```

**Review by benbp — APPROVED** [review]

**Comment by kushagraThapar** [comment]
@sima-zhu / @benbp - seems like the above azure-sdk-tools - sync - eng-common CI is stuck / or waiting for some approval. 
Can we please merge this PR soon, so that other Cosmos CI PRs can be merged, this is currently blocking our Cosmos CI, and all development / release process. Thanks!

File: eng/common/scripts/git-diff-changes.ps1
Comment (weshaggard) [inline eng/common/scripts/git-diff-changes.ps1:35]: We should put this block inside the else from above so we don't end up with extra logging in the empty case.

File: eng/common/scripts/git-diff-changes.ps1
Comment (weshaggard) [inline eng/common/scripts/git-diff-changes.ps1:19]: We probably also need an optional parameter to pass in a path filter. (i.e. for things like eng/common).

**Pull request merged**
